### PR TITLE
fix(global-search): unable to stop data object synchronization

### LINKED ISSUE #2767: [Bug]: Metabase synchronization has been initiated continuously
### ODC version

odc430bp

### OB version

*

### What happened?

image

### What did you expect to happen?

Metabase synchronization will not be initiated continuously

### How can we reproduce it (as minimally and precisely as possible)?

Enter link

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
the front-end will call sync api continuously, the reason of it is that the object sync status is 'PENDING' without changing by the time. 

I add a new property `max-pending-hours` to limit the maximum time an object's state can be kept on 'PENDING' to fix this issue.

#### Which issue(s) this PR fixes:

Fixes #2767

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java [hunk 1]
```diff
@@ -793,6 +798,14 @@ public void updateObjectLastSyncTimeAndStatus(@NotNull Long databaseId,
         databaseRepository.setObjectLastSyncTimeAndStatusById(databaseId, new Date(), status);
     }
 
+    @SkipAuthorize("odc internal usage")
+    @Transactional(rollbackFor = Exception.class)
+    public void refreshExpiredPendingDBObjectStatus() {
+        this.databaseRepository.setObjectSyncStatusByObjectSyncStatusAndObjectLastSyncTimeBefore(
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java:804]: we may add log here ,how much rows affected.

Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java:804]: done

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by PeachThinking — APPROVED** [review]
lgtm

**Pull request merged**